modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/application/resource/test/HeadlessBuilderResourceTest.java

```diff
@@ -339,6 +339,56 @@ public void testGetWithRequestFilter() throws Exception {
 			JSONCompareMode.LENIENT);
 	}
 
+	@Test
+	public void testPutByExternalReferenceCode() throws Exception {
+		_addAPIApplication(
+			_API_APPLICATION_ERC_1, _API_ENDPOINT_ERC_1, _BASE_URL_1,
+			_objectDefinition1.getExternalReferenceCode(),
+			_API_APPLICATION_PATH_1);
+
+		_publishAPIApplication(_API_APPLICATION_ERC_1);
+
+		_assertSuccessfulHttpCode(
+			HTTPTestUtil.invokeToHttpCode(
+				JSONUtil.put(
+					"applicationStatus", "unpublished"
+				).put(
+					"baseURL", "test"
+				).put(
+					"title", "title"
+				).toString(),
+				"headless-builder/applications/by-external-reference-code/" +
+					_API_APPLICATION_ERC_1,
+				Http.Method.PUT));
```

Comment:
Our `Then` would be in that case calling to the endpoint with new base URL to make sure it is a proper code status.